flow360/component/simulation/outputs/outputs.py

```diff
@@ -0,0 +1,134 @@
+"""
+
+
+class _AnimationSettings(Flow360BaseModel):
+    """
+    Controls how frequently the output files are generated.
+    """
+
+    frequency: Optional[int] = pd.Field(
+        default=-1,
+        ge=-1,
+        description="Frequency (in number of physical time steps) at which output is saved. -1 is at end of simulation.",
+    )
+    frequency_offset: Optional[int] = pd.Field(
+        default=0,
+        description="Offset (in number of physical time steps) at which output animation is started. 0 is at beginning of simulation.",
+    )
+
+
+class _TimeAverageAdditionalAnimationSettings(Flow360BaseModel):
+    """
+    Additional controls when using time-averaged output.
+
+    Notes
+    -----
+        Old `computeTimeAverages` can be infered when user is explicitly using for example `TimeAverageSurfaceOutput`.
+    """
+
+    start_step: Optional[Union[pd.NonNegativeInt, Literal[-1]]] = pd.Field(
+        default=-1, description="Physical time step to start calculating averaging"
+    )
+
+
+class SurfaceOutput(_AnimationSettings):
+    entities: EntityList[Surface] = pd.Field(alias="surfaces")
+    write_single_file: Optional[bool] = pd.Field(
+        default=False,
+        description="Enable writing all surface outputs into a single file instead of one file per surface. This option currently only supports Tecplot output format. Will choose the value of the last instance of this option of the same output type (SurfaceOutput or TimeAverageSurfaceOutput) in the `output` list.",
+    )
+    output_fields: UniqueAliasedItemList[SurfaceFields] = pd.Field()
+
+
+class TimeAverageSurfaceOutput(SurfaceOutput, _TimeAverageAdditionalAnimationSettings):
+    """
+    Caveats:
+    Solver side only accept exactly the same set of output_fields (is shared) between VolumeOutput and TimeAverageVolumeOutput.
+    """
+
+    pass
+
+
+class VolumeOutput(_AnimationSettings):
+    output_fields: UniqueAliasedItemList[VolumeFields] = pd.Field()
+
+
+class TimeAverageVolumeOutput(VolumeOutput, _TimeAverageAdditionalAnimationSettings):
+    """
+    Caveats:
+    Solver side only accept exactly the same set of output_fields (is shared) between VolumeOutput and TimeAverageVolumeOutput.
+    Also let's not worry about allowing entities here as it is not supported by solver anyway.
+    """
+
+    pass
+
+
+class SliceOutput(_AnimationSettings):
+    slices: UniqueItemList[Slice] = pd.Field()
+    output_fields: UniqueAliasedItemList[SliceFields] = pd.Field()
+
+
+class IsosurfaceOutput(_AnimationSettings):
+    isosurfaces: UniqueItemList[Isosurface] = pd.Field()
+    output_fields: UniqueAliasedItemList[CommonFields] = pd.Field()
+
+
+class SurfaceIntegralOutput(_AnimationSettings):
+    monitors: UniqueItemList[SurfaceList] = pd.Field()
+    output_fields: UniqueAliasedItemList[CommonFields] = pd.Field()
+
+
+class ProbeOutput(_AnimationSettings):
+    probes: UniqueItemList[Probe] = pd.Field()
```

Comment:
why 'Probe', not 'Point', and why probes, not entities?

Reply:
PointOutput is strange as Point is a geometrical concept. Probe is used in both Tecplot and Paraview which is more related to the monitor concept.

Reply:
I changed the probes to entities but keep probe as alias. entities is too generic I think for user to understand what they should put here.